docs/Features/XMLReferencesFeatures.md

```diff
@@ -0,0 +1,161 @@
+# XML References Features
+
+## Attribute node references
+
+Given this [docbook](https://docbook.org/) XML file sample:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.4//EN" "http://www.docbook.org/xml/4.4/docbookx.dtd">
+<book>
+    <chapter id="chapter-1">
+
+        <xref linkend="chapter-1" />
+
+    </chapter>
+
+    <chapter id="chapter-2">
+
+    </chapter>
+</book>
+```
+
+In this sample, `linkend` attribute in `<xref linkend="chapter-1" />` references the `chapter-1` declared in `<chapter id="chapter-1">`. [vscode-xml](https://github.com/redhat-developer/vscode-xml) provides a completion, definition, highlighting support to support this kind of references easily with the `xml.references` settings. For docbook case, you can declare this settings:
+
+```json
+  "xml.references": [
+    {
+      "pattern": "**/*.xml",
+      "expressions": [
+        {
+          "from": "xref/@linkend",
+          "to": "@id"
+        }
+      ]
+    }
+  ]
+```
+
+After saving this setting, you will get completion, go to definition, and highlighting for `xref` in the docbook file:
+
+![XML References with docbook](../images/Features/XMLReferencesWithDocbook.gif)
+
+Please note that [resolve external entities](https://github.com/redhat-developer/vscode-xml/blob/main/docs/Validation.md#resolve-external-entities) must be enabled for this DTD to work.
+
+The `xml.references` settings is an array of objects with the following properties:
+
+ * `pattern`:  matches the files that reference declared with `expressions` applies to. See [glob syntax](https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob) for more information about the pattern syntax. 
+ * `expressions`: array of reference expression:
+    * `prefix (optional)`: the prefix to use (ex : '#')  for from.
+    * `from`: the from reference DOM node declared with XPath.
+    * `to`: the to reference DOM node declared with XPath.
```

Comment:
Maybe add a short note that only attributes and text nodes are supported currently

Reply:
I try to fix , it and I add a short description on the top of the page. is it better?

Reply:
Yes, I think this makes a lot of sense.